src/code.jl

```diff
@@ -96,23 +96,38 @@ Base.convert(::Type{Assignment}, p::Pair) = Assignment(pair[1], pair[2])
 
 toexpr(a::Assignment, st) = :($(toexpr(a.lhs, st)) = $(toexpr(a.rhs, st)))
 
-function toexpr(O, st)
-    !istree(O) && return O
-    op = operation(O)
+function_to_expr(op, args, st) = nothing
+
+function function_to_expr(::typeof(^), O, st)
     args = arguments(O)
-    if op === (^) && length(args) == 2 && args[2] isa Number && args[2] < 0
+    if length(args) == 2 && args[2] isa Number && args[2] < 0
         ex = args[1]
         if args[2] == -1
             return toexpr(Term{Any}(inv, [ex]), st)
         else
             return toexpr(Term{Any}(^, [Term{Any}(inv, [ex]), -args[2]]), st)
         end
-    elseif op === (SymbolicUtils.ifelse)
-        return :($(toexpr(args[1], st)) ? $(toexpr(args[2], st)) : $(toexpr(args[3], st)))
-    elseif op isa Sym && haskey(st.symbolify, O)
-        return st.symbolify[O]
     end
-    return Expr(:call, toexpr(op, st), map(x->toexpr(x, st), args)...)
```

Comment:
This already did `toexpr(op, st)` why is that not sufficient?

Reply:
Do you mean to move the generic case to `toexpr`, too?